enterprise/backup/src/main/java/org/neo4j/backup/BackupCommandArgumentHandler.java

```diff
@@ -0,0 +1,172 @@
+
+    BackupCommandArgumentHandler()
+    {
+    }
+
+    public static Arguments arguments()
+    {
+        return arguments;
+    }
+
+    CheckConsistencyConfig establishConfigurationOverride( Config config ) throws IncorrectUsage
+    {
+        boolean checkGraph;
+        boolean checkIndexes;
+        boolean checkLabelScanStore;
+        boolean checkPropertyOwners;
+        try
+        {
+            // We can remove the loading from config file in 4.0
+            if ( arguments.has( "cc-graph" ) )
+            {
+                checkGraph = arguments.getBoolean( "cc-graph" );
+            }
+            else
+            {
+                checkGraph = ConsistencyCheckSettings.consistency_check_graph.from( config );
+            }
+            if ( arguments.has( "cc-indexes" ) )
+            {
+                checkIndexes = arguments.getBoolean( "cc-indexes" );
+            }
+            else
+            {
+                checkIndexes = ConsistencyCheckSettings.consistency_check_indexes.from( config );
+            }
+            if ( arguments.has( "cc-label-scan-store" ) )
+            {
+                checkLabelScanStore = arguments.getBoolean( "cc-label-scan-store" );
+            }
+            else
+            {
+                checkLabelScanStore = ConsistencyCheckSettings.consistency_check_label_scan_store.from( config );
+            }
+            if ( arguments.has( "cc-property-owners" ) )
+            {
+                checkPropertyOwners = arguments.getBoolean( "cc-property-owners" );
+            }
+            else
+            {
+                checkPropertyOwners = ConsistencyCheckSettings.consistency_check_property_owners.from( config );
+            }
+            return new CheckConsistencyConfig( checkGraph, checkIndexes, checkLabelScanStore, checkPropertyOwners );
+        }
+        catch ( IllegalArgumentException e )
+        {
+            throw new IncorrectUsage( e.getMessage() );
+        }
+    }
+
+    OnlineBackupRequiredArguments establishRequiredArguments( String... args ) throws IncorrectUsage
+    {
+        try
+        {
+            OptionalHostnamePort address = toOptionalHostnamePortFromRawAddress( arguments.parse( args ).get( "from" ) );
+            Path folder = arguments.getMandatoryPath( ARG_NAME_BACKUP_DIRECTORY );
+            String name = arguments.get( ARG_NAME_BACKUP_NAME );
+            boolean fallbackToFull = arguments.getBoolean( ARG_NAME_FALLBACK_FULL );
+            boolean doConsistencyCheck = arguments.getBoolean( ARG_NAME_CHECK_CONSISTENCY );
+            long timeout = arguments.get( ARG_NAME_TIMEOUT, TimeUtil.parseTimeMillis );
+            Optional<Path> additionalConfig = arguments.getOptionalPath( ARG_NAME_ADDITIONAL_CONFIG_DIRECTORY );
+            Path reportDir = arguments.getOptionalPath( ARG_NAME_REPORT_DIRECTORY )
+                    .orElseThrow( () -> new IllegalArgumentException( "cc-report-dir must be a path" ) );
```

Comment:
The exception message can use the relevant constant

Reply:
Addressed in next push